src/Xrefcheck/Core.hs

```diff
@@ -17,6 +17,7 @@ import Data.Default (Default (..))
 import qualified Data.List as L
```

Comment:
We should also add some basic tests for this. 

Check out the `mockServer` we have in `Test.Xrefcheck.Util`. You can use that as a reference to create a new server and use it to test this new feature.